rfcs/text/0030-linux-event-model.md

```diff
@@ -0,0 +1,1449 @@
+      platform: 'centos',
+      version: '7.9.2009'
+    }
+  },
+  process: {
+    id: '4321',
+    args: ['/bin/bash'],
+    args_count: 1,
+    command_line: 'bash',
+    executable: '/bin/bash',
+    name: 'bash',
+    interactive: true,
+    working_directory: '/home/kg',
+    pid: 3,
+    ppid: 2,
+    sid: 1,
+    pgid: 2,
+    start: '2021-10-14T08:05:34.853Z',
+    sid_start: '2021-10-14T08:05:34.853Z',
+    ppid_start: '2021-10-14T08:05:34.853Z',
+    pgid_start: '2021-10-14T08:05:34.853Z',
+    end: '2021-10-14T10:05:34.853Z',
+    exit_code: 137,
+    user: {
+      id: '2',
+      name: 'kg',
+    },
+    parent: {
+      id: '4322',
+      args: ['/bin/sshd'],
+      args_count: 1,
+      command_line: 'sshd',
+      executable: '/bin/sshd',
+      name: 'sshd',
+      interactive: true,
+      working_directory: '/',
+      pid: 2,
+      ppid: 1,
+      sid: 1,
+      pgid: 2,
+      start: '2021-10-14T08:05:34.853Z',
+      sid_start: '2021-10-14T08:05:34.853Z',
+      ppid_start: '2021-10-14T08:05:34.853Z',
+      pgid_start: '2021-10-14T08:05:34.853Z',
+      user: {
+        id: '0',
+        name: 'root',
+      },
+      fds: [
+        {
+          descriptor:0,
+          type: 'char_device',
+          char_device: {
+            major: 8,
+            minor: 1
+          }
+        }
+      ],
+      tty: {
+        descriptor: 0,
+        type: 'char_device',
+        char_device: {
+          major: 8,
+          minor: 1
+        }
+      },
+    },
+    session_leader: {
+      id: '4321',
+      args: ['bash'],
+      args_count: 1,
+      command_line: 'bash',
+      executable: '/bin/bash',
+      name: 'bash',
+      interactive: true,
+      working_directory: '/home/kg',
+      pid: 3,
+      ppid: 1,
+      sid: 1,
+      pgid: 2,
+      start: '2021-10-14T08:05:34.853Z',
+      sid_start: '2021-10-14T08:05:34.853Z',
+      ppid_start: '2021-10-14T08:05:34.853Z',
+      pgid_start: '2021-10-14T08:05:34.853Z',
+      user: {
+        id: '2',
+        name: 'kg',
+      },
+      fds: [
+        {
+          descriptor:0,
+          type: 'char_device',
+          char_device: {
+            major: 8,
+            minor: 1
+          }
+        }
+      ],
+      tty: {
+        descriptor: 0,
+        type: 'char_device',
+        char_device: {
+          major: 8,
+          minor: 1
+        }
+      },
+    },
+    entry_leader: {
+      id: '4321',
+      args: ['bash'],
+      args_count: 1,
+      command_line: 'bash',
+      executable: '/bin/bash',
+      name: 'bash',
+      interactive: true,
+      working_directory: '/home/kg',
+      pid: 3,
+      ppid: 1,
+      sid: 1,
+      pgid: 2,
+      start: '2021-10-14T08:05:34.853Z',
+      sid_start: '2021-10-14T08:05:34.853Z',
+      ppid_start: '2021-10-14T08:05:34.853Z',
+      pgid_start: '2021-10-14T08:05:34.853Z',
+      user: {
+        id: '2',
+        name: 'kg',
+      },
+      entry_meta: {
+        type: 'sshd',
+        source: {
```

Comment:
Yea, I tried to come up with something similar to that, but then we didn't have a nice spot for entry_meta.type: sshd | kublet | etc...

The nesting of this under process.entry_leader co-locates the information. It's not clear with top level 'source' that this is information related to the entry_leader not the top level "process"

Reply:
@m-sample any objections to moving source to top level? It may help surface this info for any tables that use this field. I noticed the table in the security solution "alerts" section defaults to having columns for source.ip and destination.ip. 
![image](https://user-images.githubusercontent.com/16198204/149601237-63dacd17-be4a-4a49-adbf-6640af4ef9df.png)

This could also make writing rules simpler, e.g   source.ip: 24.68.1.123 vs process.entry_leader.entry_meta.source.ip: 24.68.1.123

@ferullo Do you have any suggestions with regards to tracking remote ssh/kublet ip's. Does endpoint track client ips atm?

Reply:
`source.ip` is the field used for network events as well. What would a network event from within a container look like if `source.ip` is already "claimed" for the process leader's IP?

Reply:
Great point. Ok, probably best to keep source nested in entry_meta then, unless there are any other ideas.